profiles/common/modules/features/nexteuropa_newsroom/nexteuropa_newsroom.admin.inc

```diff
@@ -0,0 +1,597 @@
+    ),
+  );
+
+  $form['newsroom_others']['toggle_title'] = array(
+    '#type' => 'submit',
+    '#value' => 'Switch title fields',
+    '#description' => t('Switch title and name fields'),
+    '#prefix' => '<div id="newsroom_toggle_title">',
+    '#suffix' => '<div class="feedback"></div></div>',
+    '#ajax' => array(
+      'callback' => '_nexteuropa_newsroom_toggle_title',
+    ),
+  );
+
+  $form['newsroom_others']['newsroom_order_by_last_update'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Sort newsroom boxes chronologically in the summary blocks'),
+    '#default_value' => variable_get('newsroom_order_by_last_update', 0),
+  );
+
+  $form['newsroom_others']['newsroom_display_highlights_begin'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Highlighted topics are displayed at the begining of the list.'),
+    '#default_value' => variable_get('newsroom_display_highlights_begin', 1),
+  );
+
+  $form['newsroom_others']['newsroom_display_agenda'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Activate agenda functionality'),
+    '#default_value' => variable_get('newsroom_display_agenda', 1),
+  );
+
+  $form['newsroom_others']['newsroom_agenda_after_highlights'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Display Agenda after the highlights'),
+    '#default_value' => variable_get('newsroom_agenda_after_highlights', 1),
+  );
+
+  $form['newsroom_others']['newsroom_proposal_url'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Newsroom proposal URL'),
+    '#default_value' => variable_get('newsroom_proposal_url', NULL),
+  );
+
+  $vocabulary = taxonomy_vocabulary_machine_name_load(NEXTEUROPA_NEWSROOM_TYPE_VOCABULARY);
+  $types = entity_load('taxonomy_term', FALSE, array('vid' => $vocabulary->vid));
+
+  if (!empty($types)) {
+    $form['newsroom_paths'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Newsroom content path settings'),
+      '#prefix' => '<div id="newsroom_reset">',
+      '#suffix' => '</div>',
+      '#description' => t("You can define here a root path for your items basing on the item type. The default behavior is to have sections correponding to the types, so if you have two types like
+            news and events you will get pattern like news/[node:title], events/[node:title], you can override the root path for your elements here. Do not use leading or trailing slashes"),
+      '#collapsible' => TRUE,
+      '#collapsed' => TRUE,
+      '#access' => user_access('administer newsroom settings'),
+    );
+
+    $form['newsroom_paths']['patterns'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Do you want to take over the configuration of pattern for node aliases?'),
+      '#description' => t('if this is unchecked the newsroom will not alter the aliases for its nodes.'),
+      '#default_value' => variable_get('newsroom_patterns', FALSE),
+    );
+
+    foreach ($types as $tid => $type) {
```

Comment:
Where it was possible I have replaced the code. So, we call the same method to get types and topics.